### ISSUE TITLE
Should block('*') apply to elems?

### ISSUE THREAD

**Issue by miripiruni** [bem/bem-xjst#256]
Examples: http://goo.gl/SBJszu

**Comment by miripiruni** [comment]
cc @veged @mishanga @st-lukas

**Comment by miripiruni** [comment]
I think no. Current behaviour looks like a bug.

**Comment by miripiruni** [comment]
In the other hand, this behaviour is quite comfortable for those who want to match to all entities (blocks and elems).

**Comment by zxqfox** [comment]
Of course, not! Whenever we need to apply something to every element we can use: `block('*').elem('*')`.

**Comment by St-Lukas** [comment]
Current behaviour is surprising when you expect that similar templates apply only to blocks.

**Comment by mishanga** [comment]
Поведение действительно странное, но его нельзя просто так оторвать, т.к. придется дублировать тело шаблона для `block('*')` и `block('*').elem('*')`.
Нужен какой-то более универсальный подпредикат, под который будут попадать вообще все узлы bemjson-дерева.

**Comment by veged** [comment]
особенно это будет проблемой в https://nda.ya.ru/3RWXFK — нужно придумывать альтернативный способ

**Comment by St-Lukas** [comment]
+1 за более универсальный подпредикат, он как раз и поможет в случае с https://nda.ya.ru/3RWXFK
Вот еще пример, почему так оставлять не стоит:
https://goo.gl/5ILcm8
Т.е. приоритет для элемента у ``block('*')`` выше, чем у ``elem('*')``

**Comment by miripiruni** [comment]
TO ALL: А чем вот так плохо? https://goo.gl/ABfIlj

**Comment by zxqfox** [comment]
https://goo.gl/ygm5Tf

Должно ли это работать для `{justASimpleObject: true}`?

**Comment by miripiruni** [comment]
@zxqfox Nope

**Comment by zxqfox** [comment]
Если 

> Нужен какой-то более универсальный подпредикат, под который будут попадать вообще все узлы bemjson-дерева.

и

> > Должно ли это работать для `{justASimpleObject: true}`?

> @zxqfox Nope

то:

```js
any(block('*'), block('*').elem('*')).tag()('span') // как бы мультиплицирует
```
матчится на все узлы **BEMJSON**-дерева, но не матчится на `{justAnObject: true}`, но хочет введения подпредиката `any`, который решает задачу логического `or` для матчеров (либо-либо).
